### PR TITLE
Fix segfault in machine.go

### PR DESCRIPTION
**Pull request by nhorman**
If the Open call in PowerSet fails, then the pbnj server will never set
its metadata field, and consequently, without checking the error code,
we then call client.GetMetadata, that element is dereferenced, leading
to a segfault.

Check the error code

Signed-off-by: Neil Horman <neil.horman@privafy.com>

### CODE REVIEW

File: grpc/oob/machine/machine.go [hunk 1]
```diff
@@ -254,6 +254,12 @@ func (m Action) PowerSet(ctx context.Context, action string) (result string, err
 	client := bmclib.NewClient(host, "623", user, password, bmclib.WithLogger(m.Log))
 
 	err = client.Open(ctx)
+	if err != nil {
```
Comment (jacobweinstock) [inline grpc/oob/machine/machine.go:257]: Looks like this duplicates some lines ([L267-L275](https://github.com/tinkerbell/pbnj/pull/111/files#diff-58e9fa0e549a798eb5175bffdc3b82999c98e43ac5ada9340d5b2f93babb117aR267-R275)) below. Would you be open to maybe using those lines here instead? We'd probably still want to get the span and status message.

**Review by jacobweinstock — CHANGES_REQUESTED** [review]
Hey @nhorman, thanks for catching this!

File: grpc/oob/machine/machine.go (hunk 1 above)
Comment (nhorman) [reply grpc/oob/machine/machine.go:257]: sure

Comment (nhorman) [reply grpc/oob/machine/machine.go:257]: done

Comment (jacobweinstock) [reply grpc/oob/machine/machine.go:257]: thanks! looks like we are still duplicating the `err != nil` check block from ([L267-L275](https://github.com/tinkerbell/pbnj/pull/111/files#diff-58e9fa0e549a798eb5175bffdc3b82999c98e43ac5ada9340d5b2f93babb117aR267-R275)). Maybe remove those lines? Other options? What do you think?

**Review by jacobweinstock — CHANGES_REQUESTED** [review]

**Comment by jacobweinstock** [comment]
Wanted to thank you again for this report and let you know I also just [landed a change](https://github.com/bmc-toolbox/bmclib/pull/246) in the backing [bmclib](https://github.com/bmc-toolbox/bmclib) library to address this.

File: grpc/oob/machine/machine.go (hunk 1 above)
Comment (jacobweinstock) [reply grpc/oob/machine/machine.go:257]: Hey @nhorman, following up, I think with this error handling now in the more correct place we would want to remove the unneeded lines below: [270-278](https://github.com/tinkerbell/pbnj/blob/ddb83c5fa7f665d623e992f7ca69ad42e137c685/grpc/oob/machine/machine.go#L270-L278).

**Review by jacobweinstock — CHANGES_REQUESTED** [review]

**Comment by nhorman** [comment]
sorrry, lost track of this, will get to it shortly

**Comment by nhorman** [comment]
updated, removed duplicate error path

**Review by jacobweinstock — APPROVED** [review]
Thank you for this contribution!

**Pull request merged**